### DISCUSSION TITLE
QTable with per-row context menu without using body slot

### DISCUSSION THREAD

**Discussion by davthedev** [quasarframework/quasar#9428] [General - Components / Directives / etc]
Hi,

I would like to show contextual operations for each row of a QTable, using a right-click menu.

So far, the only successful solution I came up with is to use the body slot and add the QMenu there.
The drawback of that approach are:
- I cannot display the checkboxes to select items as the whole row is overridden.
- Automatic column mapping from the model is disabled

Is there any other possible approach which does not require overriding the body slot?

I have tried this idea:
- Declare the QMenu next to the QTable
- Add a listener to the row-contextmenu event
- Programmatically show the menu in reaction to that event, which I failed to make it work so far

**Comment by ontwikkelfabriek** [comment]
@davthedev 
> I cannot display the checkboxes to select items as the whole row is overridden.

Yes you can , you can put your own selection checkboxes in the q-table's selection slot.

> Automatic column mapping from the model is disabled

What do you mean?

**Reply by davthedev** [reply]
By "automatic", I mean the default QTable behaviour where columns are defined using the `columns` property.
By overriding the body slot, I need to maintain both the row model and its whole representation with q-td's, which is less reusable. I can live with it but my idea is to make this behaviour generic in multiple parts of my project for different data sets, while only passing the `data` and `columns` properties.

Regarding the checkboxes, I would like to reuse the default template, as if I did not pass the body slot customization in the first place.
